File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        prev=None
        self.mini=float('inf')
        def inorder(root):
            nonlocal prev
            if not root:
                return
            inorder(root.left)
            if  prev:
                self.mini=min(self.mini,root.val-prev.val)
            prev=root
            #mini=min()
            inorder(root.right)
        inorder(root)
        return self.mini

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private TreeNode prev=null;
    private int diff=Integer.MAX_VALUE;
    public int getMinimumDifference(TreeNode root) {
        inorder(root);
        return diff;
    }
    private void inorder(TreeNode root){
        if (root==null){
            return;
        }
        inorder(root.left);
        if (prev!=null){
            diff=Math.min(diff,root.val-prev.val);
        }
        prev=root;
        inorder(root.right);
    }
}